android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java

```diff
@@ -127,8 +128,24 @@ public String getName() {
     @ReactMethod
     public void addAndroidListener(String eventName) {
         Logger.info("UrbanAirshipReactModule - Event listener added: " + eventName);
-        listenerCount++;
-        EventEmitter.shared().setEnabled(getReactApplicationContext(), true);
+
+        synchronized (EventEmitter.shared().knownListeners) {
```

Comment:
this code should live in EventEmitter

Reply:
I would just call through to event emitter: `EventEmitter.shared().addListener(eventName)`